sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVParser.scala

```diff
@@ -28,11 +28,10 @@ import org.apache.spark.internal.Logging
  * Read and parse CSV-like input
  *
  * @param params Parameters object
- * @param headers headers for the columns
  */
-private[sql] abstract class CsvReader(params: CSVOptions, headers: Seq[String]) {
+private[sql] class CsvReader(params: CSVOptions) {
 
-  protected lazy val parser: CsvParser = {
+  private lazy val parser: CsvParser = {
```

Comment:
Oh, it seems it doesn't have to be. I will remove this and take a close look again.